### PR TITLE
clean ci

### PR DESCRIPTION
**Pull request by ruslandoga**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,29 +7,22 @@ on:
...
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-24.04
 
     env:
       MIX_ENV: test
 
     strategy:
       # https://hexdocs.pm/elixir/compatibility-and-deprecations.html#between-elixir-and-erlang-otp
       matrix:
-        os: [ubuntu-22.04, ubuntu-24.04]
```
Comment (ruslandoga) [inline .github/workflows/ci.yml:18]: Erlang / Elixir don't depend on the OS. Also this was overwriting the caches.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,29 +7,22 @@ on:
...
             otp: 27
         exclude:
           - elixir: 1.14
             otp: 26
-          - elixir: 1.14
-            otp: 27
-          - elixir: 1.15
-            otp: 27
-          - elixir: 1.16
-            otp: 27
```
Comment (ruslandoga) [inline .github/workflows/ci.yml:32]: Elixir 1.17 only works with OTP 27. It's easier to include that pair once rather than exclude all the others.

**Comment by ruslandoga** [comment]
Seems like `Hammer.ETS.CleanTest` is flaky and needs improvement.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (epinault) [reply .github/workflows/ci.yml:18]: I am well aware of this , but was nice to have predictable LTS. I can remove this

File: .github/workflows/ci.yml (hunk 2 above)
Comment (epinault) [reply .github/workflows/ci.yml:32]: yup, hence why I had the matrix I had. I am not sure I like the include vs exclude in term of readability but I ll go with it for now

File: .github/workflows/ci.yml (hunk 1 above)
Comment (epinault) [reply .github/workflows/ci.yml:18]: I know they dont, but getting predictable LTS helps sometime . I have had compiling issue in the past still for nif and stuff

**Comment by epinault** [comment]
yea was going to say we have a flaky test. we can try to fix it later

**Review by epinault — APPROVED** [review]

**Pull request merged**